src/doc/rustc/src/codegen-options/index.md

```diff
@@ -435,6 +435,10 @@ Equivalent to the "uppercase" `-fPIC` or `-fPIE` options in other compilers,
 depending on the produced crate types.  \
 This is the default model for majority of supported targets.
 
+- `pie` - position independent executable, relocatable code but without support for symbol
+interpositioning. Equivalent to the "uppercase" `-fPIE` option in other compilers. `pie`
+code cannot be linked into shared libraries.
```

Comment:
What happens if you attempt to link `pie` code into a shared library? Do you get a linker error or does this result in some kind of failure at runtime? 

I'm wondering if we should try to detect this situation when building dylibs or executables and provide a better error.

Reply:
This will result in a compiler error in any case where `-fpie` would change the generated object files in the first place.

```
$ echo "int foo = 42; int import() { return foo; } int export() { return import(); }" | gcc -fpie -xc -c - -o a.o
$ gcc -fPIC -shared a.o -o a.so
/usr/bin/ld: a.o: relocation R_X86_64_PC32 against symbol `foo' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: nonrepresentable section on output
collect2: error: ld returned 1 exit status
```